### PR TITLE
No wait for search shards in reserved role mapping updates

### PR DESCRIPTION
**Pull request by n1v0lg**
Sets `wait_for_active_shards` to `NONE` for indexing role mappings. Previously, this was set to the default value of `1` which in stateless implies waiting to an active search shard. However, we only need to wait for the primary. For indexing, we implicitly wait for the primary shard to be available (both in stateless and stateful) so it's safe to use `NONE` here instead.

### CODE REVIEW

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/support/mapper/NativeRoleMappingStore.java
Comment (henningandersen) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/support/mapper/NativeRoleMappingStore.java:246]: Do we need to pass the information through the `PutRoleMappingRequest`? It seems like we might as well use `NONE` here always. Seems good also for other uses of the store.

We always wait for the primary anyway during indexing.

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/support/mapper/NativeRoleMappingStore.java
Comment (henningandersen) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/support/mapper/NativeRoleMappingStore.java:279]: Same comment as above.

**Review by henningandersen — COMMENTED** [review]
This looks good to me, though I think we can simplify and you probably need a security team approval on it.

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/support/mapper/NativeRoleMappingStore.java
Comment (n1v0lg) [reply x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/support/mapper/NativeRoleMappingStore.java:246]: Sorry after I opened this PR, I actually thought of not including it on the request (to avoid unnecessary serialization changes) and simply passing it through as a separate param to `putRoleMapping`. 

> It seems like we might as well use NONE here always. Seems good also for other uses of the store. We always wait for the primary anyway during indexing. 

Just to validate: this is the case for stateful also? This API is exposed to stateful customers.

Comment (n1v0lg) [reply x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/support/mapper/NativeRoleMappingStore.java]: I've pushed the suggested change; provided we wait for the primary on indexing in stateful, this change makes perfect sense to me.

**Review by jakelandis — APPROVED** [review]
LGTM

**Comment by n1v0lg** [comment]
@elasticmachine update branch

**Pull request merged**
